policies/releasestrat.html

```diff
@@ -55,65 +56,20 @@ <h5>
 	  fixes. Before that, bug and security fixes will be applied
 	  as appropriate.</p>
 
-	  <p>The next version of OpenSSL will be 1.1.1 which will be an LTS release.
-	  This is currently in development and has a primary focus of implementing
-	  TLSv1.3. The RFC for TLSv1.3 has not yet been published by the IETF.
-	  OpenSSL 1.1.1 will not have its final release until that has happened.</p>
-
-	  <p>The draft release timetable for 1.1.1 is as follows. This may be
-          amended at any time as the need arises.</p>
-
+	  <p>The following stability rules apply to all changes made to code
+	  targetted for a version after 1.1.1:</p>
 	  <ul>
-	    <li>13th February 2018, alpha release 1 (pre1)</li>
-	    <li>27th February 2018, alpha release 2 (pre2)</li>
-	    <li>20th March 2018, beta release 1 (pre3)
-              <ul>
-	        <li>OpenSSL_1_1_1-stable created (feature freeze)</li>
-	        <li>master becomes basis for 1.1.2 or 1.2.0 (TBD)</li>
-	      </ul>
-	    <li>3rd April 2018, beta release 2 (pre4)</li>
-	    <li>17th April 2018, beta release 3 (pre5)</li>
-	    <li>1st May 2018, beta release 4 (pre6)</li>
-	    <li>29th May 2018, beta release 5 (pre7)</li>
-	    <li>19th June 2018, beta release 6 (pre8)</li>
-	    <li>Release readiness check following pre8 release (new release
-	        cycles added if required)</li>
+	    <li>No existing public interface can be removed or modified until its
```

Comment:
We need to define what we mean by "public interface" here clearly as there have been rather different views on that.
The two views I have seen regularly expressed:
1) in the header files that are installed
2) in the pod documentation files that are installed

Clearly 2) is a much smaller list than 1).

Reply:
In my mind it is clearly (1)

Reply:
It's actually a problem that (2) is smaller than (1). Lots of things still need documentation.

Reply:
FWIW, I remember trying to remove something because it wasn't documented and Tim was the leading proponent of "it's in the header file" but that most of the team agreed and I was definitely in the rough. 
```
; ./util/find-doc-nits -u
# Duplicate i2d_PublicKey in doc/man3/d2i_X509.pod and doc/man3/d2i_PrivateKey.pod
# Found 4577 in util/libcrypto.num
# Found 1607 missing from util/libcrypto.num

# Found 499 in util/libssl.num
# Found 57 missing from util/libssl.num

# Checking macros (approximate)
# Found 229 macros missing (not all should be documented)
; 
```

Reply:
Try the -d flag also.

Reply:
So working those rules taking the function DES_cbc_encrypt as an example - it is in the header file des.h and it is not currently marked deprecated. Do we consider the EVP-level API a replacement?
And in order to remove the API we have to mark it as deprecated first in a release and then wait at least 5 more years before it is removed which can only happen in a API-breaking release itself?

Reply:
Yes. Although we could (potentially) move it to a "libcrypto-legacy" API. The structure could be made opaque (according to the rules stated for making things opaque). And the internal implementation could be changed to call EVP.

Reply:
Functions that have always been documented as being not guaranteed to be API-stable would presumably get an exception (or, if later documented that way, after the same timeout as for removal).

Some other projects I work with have the luxury of an explicit export symbol list to be clear about what symbols are in the public API (intersected with the public headers, since in some cases internal symbols from one library are exposed to other libraries in the same implementation but not to the public).  Our export list is generated from the headers at present, so that just leaves us with the public headers alone, even if that leaves us in an awkward position regarding some APIs that really ought not to have existed.

Reply:
I put some text in to define public interface as "any function or macro declared in a public header file"